backend/tests/test_tokenlist.py

```diff
@@ -0,0 +1,197 @@
+            "keywords": ["mockfi"],
+            "timestamp": "2022-06-19T11:08:16.098+00:00",
+            "tokens": [
+                {"chainId": 1, "address": "0x6d6f636b5f31"},
+                {"chainId": 1, "address": "0x6d6f636b5f32"},
+            ],
+        }
+
+
+class TestTokenList:
+    log_resp = ""
+
+    async def mock_get_uniswap(self, _: str):
+        return MockResponse()
+
+    @mark.asyncio
+    async def test_get_token_list_success(self, monkeypatch: MonkeyPatch):
+        monkeypatch.setattr(log, "get_task_logger", Logger("mock_task_logger"))
+
+        monkeypatch.setattr(AsyncClient, "get", self.mock_get_uniswap)
+        monkeypatch.setattr(
+            "backend.app.libs.tokenlists.store_token_whitelist",
+            lambda x: None,
+            raising=True,
+        )
+        await get_token_list("https://tokens.coingecko.com/uniswap/all.json")
+        assert True
+
+    @mark.asyncio
+    @mark.parametrize(
+        "json_resp",
+        [
+            lambda _: {
+                "name": "MockGecko",
+                "logoURI": "https://www.mockgecko.com/mock/path/thumb.jpg",
+                "keywords": ["mockfi"],
+                "timestamp": "2022-06-19T11:08:16.098+00:00",
+                "tokens": [
+                    {"chainId": 1, "address": "0x6d6f636b5f31"},
+                    {"chainId": 1},
+                ],
+            },
+            lambda _: {
+                "name": "MockGecko",
+                "logoURI": "https://www.mockgecko.com/mock/path/thumb.jpg",
+                "keywords": ["mockfi"],
+                "timestamp": "2022-06-19T11:08:16.098+00:00",
+                "tokens": [
+                    {"address": "0x6d6f636b5f31"},
+                    {"chainId": 1, "address": "0x6d6f636b5f32"},
+                ],
+            },
+            lambda _: {
+                "name": "MockGecko",
+                "logoURI": "https://www.mockgecko.com/mock/path/thumb.jpg",
+                "keywords": ["mockfi"],
+                "timestamp": "2022-06-19T11:08:16.098+00:00",
+                "toekns": [
+                    {"chainId": 1, "address": "0x6d6f636b5f31"},
+                    {"chainId": 1, "address": "0x6d6f636b5f32"},
+                ],
+            },
+        ],
+    )
+    async def test_get_token_list_bad_resp_json(
+        self, monkeypatch: MonkeyPatch, json_resp
+    ):
+        def mock_error_log(_, error_str: str, arg):
+            self.log_resp = error_str.replace("%s", arg)
+
+        async def async_return(*_):
+            return MockResponse()
+
+        monkeypatch.setattr(Logger, "error", mock_error_log, raising=True)
+
+        url = "https://tokens.coingecko.com/uniswap/all.json"
+        monkeypatch.setattr(AsyncClient, "get", async_return)
+        monkeypatch.setattr(MockResponse, "json", json_resp)
+        await get_token_list(url)
+
+        assert (
+            self.log_resp == f"the token list url given is likely not supported: {url}"
```

Comment:
This is a useful test, but testing against logs is, I believe, a unnecessarily complicated approach. We should be trying to leverage Python exceptions as much as possible. See my other comment.

Reply:
This has been done for the latest commit, thanks 👍.